x/valset-pref/msg_server_test.go

```diff
@@ -124,3 +124,82 @@ func (suite *KeeperTestSuite) TestSetValidatorSetPreference() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestDelegateToValidatorSet() {
+	suite.SetupTest()
+
+	// prepare validators to delegate to
+	preferences := suite.PrepareDelegateToValidatorSet()
+
+	tests := []struct {
+		name           string
+		delegator      sdk.AccAddress
+		coin           sdk.Coin
+		expectedShares []sdk.Dec
+		expectPass     bool
+		valSetExists   bool
+	}{
+		{
```

Comment:
Ditto with the comment above regarding jailed validator. Consider adding a test case on it as well

Reply:
hmm not sure how i can create a test scenario where a validator is jailed and user tries to delegate to that validator  🤔

Reply:
So i did some digging around and found out that "It is possible to delegate to a jailed validator, the only difference being it will not be added to the power index until it is unjailed" therefore, i dont think we can add a very indepth test

Reply:
how about unbonded or unbonding validators?

Reply:
Here's what i found; 
1. unbonded: Validators will be removed from active set but **still can receive delegations**
2. unbonding: Validators leaves active set either through slashing, jailing etc, **all delegations can be unbonded but will have to wait for unbonding time to exhaust** 

I am hoping to test all these in either e2e or testnet